snare/tests/test_cloner_fetch_data.py

```diff
@@ -0,0 +1,96 @@
+import asyncio
+import os
+import shutil
+import sys
+import unittest
+from unittest.mock import patch
+
+import aiohttp
+from pyppeteer import launch
+from pyppeteer.errors import BrowserError
+import yarl
+
+from snare.cloner import BaseCloner, HeadlessCloner, SimpleCloner
+from snare.utils.asyncmock import AsyncMock
+from snare.utils.page_path_generator import generate_unique_path
+
+
+class TestCloner(unittest.TestCase):
+    def setUp(self):
+        self.url = yarl.URL("http://example.com")
+        self.main_page_path = generate_unique_path()
+        os.makedirs(self.main_page_path)
+        self.expected_new_url = yarl.URL("http://example.com")
+        self.expected_err_url = yarl.URL("http://example.com/status_404")
+        self.max_depth = sys.maxsize
+        self.css_validate = False
+        self.loop = asyncio.get_event_loop()
+        self.base_handler = BaseCloner(self.url, self.max_depth, self.css_validate)
+        self.simple_handler = SimpleCloner(self.url, self.max_depth, self.css_validate)
+        self.headless_handler = HeadlessCloner(self.url, self.max_depth, self.css_validate)
+
+    def test_basecloner_fetch_data(self):
+        async def test():
+            with self.assertRaises(NotImplementedError):
+                await self.base_handler.fetch_data(None, self.url, 0, 0)
+
+        self.loop.run_until_complete(test())
+
+    def test_simplecloner_fetch_data(self):
```

Comment:
what are you testing here? what is the purpose of this test? what is expected behaviour? Right now it looks for me not like an unittest test, but rather like just running the code

Reply:
I have now removed some of the tests from `fetch_data` as the function itself is pretty straight-forward and the tests for `get_body` already cover its general use-case.